### PR TITLE
[RESTEASY-1513]

### PR DESCRIPTION
**Pull request by ronsigal**
Deprecate resteasy-jackson-provider.

### CODE REVIEW

File: docbook/reference/en/en-US/modules/Json.xml [hunk 1]
```diff
@@ -1,63 +1,59 @@
...
+    currently deprecated, and will be removed in a release subsequent to 3.1.0.Final. The resteasy-jackson2-provider
+    module is based on Jackson 2. 
     </para>
+    
     <sect1>
-        <title>Using Jackson 1.9.x Outside of JBoss AS7</title>
-        <para>If you're deploying Resteasy outside of JBoss AS7 add the resteasy jackson provder to your WAR
-        pom.xml build</para>
+        <title>Using Jackson 1.9.x Outside of Wildfly 10 or EAP 7</title>
+        <para>If you're deploying Resteasy outside of Wildfly 10 or EAP 7,
```
Comment (asoldano) [inline docbook/reference/en/en-US/modules/Json.xml]: Minor note: I would stay general here and say "WildFly or JBoss EAP", without versions.

Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Json.xml]: Well, I thought of that, and I'm in favor of being as general as possible. I'm not sure if the sentence applies to, say, Wildfly 8 or EAP 6. Or EAP 5. Or Wildfly 17. ;-)

Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Json.xml]: Actually, I'm thinking maybe we shouldn't even mention EAP in the community user guide.

Comment (asoldano) [inline docbook/reference/en/en-US/modules/Json.xml]: Yeah, that makes sense

Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Json.xml]: re: "Yeah, that makes sense"

The versions, or the reference to EAP?

Comment (asoldano) [inline docbook/reference/en/en-US/modules/Json.xml]: I mean completly removing EAP references (and anyway mentioning WildFly without version if that's not strictly required)

Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Json.xml]: ++1

Comment (ronsigal) [inline docbook/reference/en/en-US/modules/Json.xml]: Ok, I removed references to EAP. Also, I referred to Wildfly 8 or Wildfly 9 or above, as appropriate.

**Pull request merged**